src/lib.rs

```diff
@@ -76,6 +77,15 @@ impl ParseOptions {
         self.recursion_limit = Some(NonZeroU32::new(limit).expect("Recursion limit must be > 0"));
         self
     }
+    /// Set the limit on the size of the substitutions table during the parsing
+    /// phase. A low limit will cause valid symbols to be rejected, but a high
+    /// limit may allow pathological symbols to run away and use an inordinate
+    /// amount of CPU and memory during parsing.
+    /// The default value is unlimited.
```

Comment:
I wonder what a good default here might be. How about `symbol.len()`, as at most one character from the symbol should insert things into the substitution table. However as the original issue mentioned, the recursive descent parsing and try-next-pattern on failure means that failure paths can add tons of substitutions and this can grow larger than the symbol length.
I’m not sure how the parser design could trigger this case even for valid symbols. Maybe 2*length in that case might be a more realistic default?